src/main/java/org/skriptlang/skript/bukkit/itemcomponents/equippable/EquippableWrapper.java

```diff
@@ -0,0 +1,241 @@
+		}
+		COMPONENT_MODEL_METHOD = componentModelMethod;
+		BUILDER_MODEL_METHOD = builderModelMethod;
+	}
+
+	public EquippableWrapper(ItemStack itemStack) {
+		super(itemStack);
+	}
+
+	public EquippableWrapper(ItemSource<?> itemSource) {
+		super(itemSource);
+	}
+
+	public EquippableWrapper(Equippable component) {
+		super(component);
+	}
+
+	public EquippableWrapper(Builder builder) {
+		super(builder);
+	}
+
+	@Override
+	public Valued<Equippable> getDataComponentType() {
+		return DataComponentTypes.EQUIPPABLE;
+	}
+
+	@Override
+	protected Equippable getComponent(ItemStack itemStack) {
+		Equippable equippable = itemStack.getData(DataComponentTypes.EQUIPPABLE);
+		if (equippable != null)
+			return equippable;
+		return Equippable.equippable(EquipmentSlot.HEAD).build();
+	}
+
+	@Override
+	protected Builder getBuilder(ItemStack itemStack) {
+		Equippable equippable = itemStack.getData(DataComponentTypes.EQUIPPABLE);
+		if (equippable != null)
+			return equippable.toBuilder();
+		return Equippable.equippable(EquipmentSlot.HEAD);
+	}
+
+	@Override
+	protected void setComponent(ItemStack itemStack, Equippable component) {
+		itemStack.setData(DataComponentTypes.EQUIPPABLE, component);
+	}
+
+	@Override
+	protected Builder getBuilder(Equippable component) {
+		return component.toBuilder();
+	}
+
+	@Override
+	public EquippableWrapper clone() {
+		EquippableWrapper clone = newWrapper();
+		Equippable base = getComponent();
+		clone.applyComponent(clone(base.slot()));
+		return clone;
+	}
+
+	/**
+	 * Returns a cloned {@link Equippable} of this {@link EquippableWrapper} with a new {@link EquipmentSlot}.
+	 */
+	public Equippable clone(EquipmentSlot slot) {
+		Equippable base = getComponent();
+		Builder builder = Equippable.equippable(slot)
+			.allowedEntities(base.allowedEntities())
+			.cameraOverlay(base.cameraOverlay())
+			.damageOnHurt(base.damageOnHurt())
+			.dispensable(base.dispensable())
+			.equipSound(base.equipSound())
+			.swappable(base.swappable());
+		setModel(builder, getModel());
+		if (HAS_EQUIP_ON_INTERACT) {
+			builder.equipOnInteract(base.equipOnInteract());
+		}
+		if (HAS_CAN_BE_SHEARED) {
+			builder.canBeSheared(base.canBeSheared());
+		}
+		if (HAS_SHEAR_SOUND) {
+			builder.shearSound(base.shearSound());
+		}
+		return builder.build();
+	}
+
+	@Override
+	public Equippable newComponent() {
+		return newBuilder().build();
+	}
+
+	@Override
+	public Builder newBuilder() {
+		return Equippable.equippable(EquipmentSlot.HEAD);
+	}
+
+	@Override
+	public EquippableWrapper newWrapper() {
+		return newInstance();
+	}
+
+	/**
+	 * Returns a {@link Collection} of {@link EntityType}s that are allowed to wear with this {@link EquippableWrapper}.
+	 */
+	public Collection<EntityType> getAllowedEntities() {
+		return getAllowedEntities(getComponent());
+	}
+
+	/**
+	 * Updates the model/assetId of this {@link EquippableWrapper} with {@code key}.
+	 * @param key The {@link Key} to set to.
+	 */
+	public void setModel(Key key) {
```

Comment:
should this be `setAssetId` to align with the API (goes for overloads below too)?